### ISSUE TITLE
Python 3.10.0b1 asyncio.get_event_loop() DeprecationWarning: There is no current event loop

### ISSUE THREAD

**Issue by hrnciar** [sqlalchemy/sqlalchemy#6537]
**Describe the bug**

asyncio.get_event_loop() is deprecated since Python 3.10.0b1. It should be either filtered out or replaced with asyncio.get_running_loop().

**To Reproduce**
Build sqlalchemy with Python 3.10.0b1.

**Error**
```
______________________ TestAsyncioCompat.test_async_error ______________________
[gw1] linux -- Python 3.10.0 /usr/bin/python3
Traceback (most recent call last):
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/testing/plugin/pytestplugin.py", line 783, in decorate
    asyncio._run_coroutine_function(fn, *args, **kwargs)
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/testing/asyncio.py", line 33, in _run_coroutine_function
    return _util_async_run_coroutine_function(fn, *args, **kwargs)
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/util/_concurrency_py3k.py", line 155, in _util_async_run_coroutine_function
    loop = asyncio.get_event_loop()
DeprecationWarning: There is no current event loop
______________ TestAsyncioCompat.test_await_fallback_no_greenlet _______________
[gw1] linux -- Python 3.10.0 /usr/bin/python3
Traceback (most recent call last):
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/base/test_concurrency_py3k.py", line 60, in test_await_fallback_no_greenlet
    await_fallback(to_await)
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/util/_concurrency_py3k.py", line 77, in await_fallback
    loop = asyncio.get_event_loop()
DeprecationWarning: There is no current event loop
________________ TestAsyncioCompat.test_await_only_no_greenlet _________________
[gw1] linux -- Python 3.10.0 /usr/bin/python3
Traceback (most recent call last):
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/testing/plugin/pytestplugin.py", line 783, in decorate
    asyncio._run_coroutine_function(fn, *args, **kwargs)
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/testing/asyncio.py", line 33, in _run_coroutine_function
    return _util_async_run_coroutine_function(fn, *args, **kwargs)
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/util/_concurrency_py3k.py", line 155, in _util_async_run_coroutine_function
    loop = asyncio.get_event_loop()
DeprecationWarning: There is no current event loop
__________________________ TestAsyncioCompat.test_ok ___________________________
[gw1] linux -- Python 3.10.0 /usr/bin/python3
Traceback (most recent call last):
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/testing/plugin/pytestplugin.py", line 783, in decorate
    asyncio._run_coroutine_function(fn, *args, **kwargs)
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/testing/asyncio.py", line 33, in _run_coroutine_function
    return _util_async_run_coroutine_function(fn, *args, **kwargs)
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/util/_concurrency_py3k.py", line 155, in _util_async_run_coroutine_function
    loop = asyncio.get_event_loop()
DeprecationWarning: There is no current event loop
______________________ TestAsyncioCompat.test_sync_error _______________________
[gw1] linux -- Python 3.10.0 /usr/bin/python3
Traceback (most recent call last):
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/testing/plugin/pytestplugin.py", line 783, in decorate
    asyncio._run_coroutine_function(fn, *args, **kwargs)
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/testing/asyncio.py", line 33, in _run_coroutine_function
    return _util_async_run_coroutine_function(fn, *args, **kwargs)
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/util/_concurrency_py3k.py", line 155, in _util_async_run_coroutine_function
    loop = asyncio.get_event_loop()
DeprecationWarning: There is no current event loop
_________________ TestAsyncioCompat.test_await_fallback_error __________________
[gw0] linux -- Python 3.10.0 /usr/bin/python3
Traceback (most recent call last):
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/testing/plugin/pytestplugin.py", line 783, in decorate
    asyncio._run_coroutine_function(fn, *args, **kwargs)
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/testing/asyncio.py", line 33, in _run_coroutine_function
    return _util_async_run_coroutine_function(fn, *args, **kwargs)
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/util/_concurrency_py3k.py", line 155, in _util_async_run_coroutine_function
    loop = asyncio.get_event_loop()
DeprecationWarning: There is no current event loop
___________________ TestAsyncioCompat.test_await_only_error ____________________
[gw0] linux -- Python 3.10.0 /usr/bin/python3
Traceback (most recent call last):
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/testing/plugin/pytestplugin.py", line 783, in decorate
    asyncio._run_coroutine_function(fn, *args, **kwargs)
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/testing/asyncio.py", line 33, in _run_coroutine_function
    return _util_async_run_coroutine_function(fn, *args, **kwargs)
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/util/_concurrency_py3k.py", line 155, in _util_async_run_coroutine_function
    loop = asyncio.get_event_loop()
DeprecationWarning: There is no current event loop
______________________ TestAsyncioCompat.test_contextvars ______________________
[gw0] linux -- Python 3.10.0 /usr/bin/python3
Traceback (most recent call last):
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/testing/plugin/pytestplugin.py", line 783, in decorate
    asyncio._run_coroutine_function(fn, *args, **kwargs)
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/testing/asyncio.py", line 33, in _run_coroutine_function
    return _util_async_run_coroutine_function(fn, *args, **kwargs)
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/util/_concurrency_py3k.py", line 155, in _util_async_run_coroutine_function
    loop = asyncio.get_event_loop()
DeprecationWarning: There is no current event loop
_____________________ TestAsyncioCompat.test_require_await _____________________
[gw0] linux -- Python 3.10.0 /usr/bin/python3
Traceback (most recent call last):
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/testing/plugin/pytestplugin.py", line 783, in decorate
    asyncio._run_coroutine_function(fn, *args, **kwargs)
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/testing/asyncio.py", line 33, in _run_coroutine_function
    return _util_async_run_coroutine_function(fn, *args, **kwargs)
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/util/_concurrency_py3k.py", line 155, in _util_async_run_coroutine_function
    loop = asyncio.get_event_loop()
DeprecationWarning: There is no current event loop
_____________ TextSyncDBAPI.test_sync_driver_execution[<lambda>1] ______________
[gw1] linux -- Python 3.10.0 /usr/bin/python3
Traceback (most recent call last):
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/testing/plugin/pytestplugin.py", line 783, in decorate
    asyncio._run_coroutine_function(fn, *args, **kwargs)
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/testing/asyncio.py", line 33, in _run_coroutine_function
    return _util_async_run_coroutine_function(fn, *args, **kwargs)
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/util/_concurrency_py3k.py", line 155, in _util_async_run_coroutine_function
    loop = asyncio.get_event_loop()
DeprecationWarning: There is no current event loop
___________________ TextSyncDBAPI.test_sync_driver_run_sync ____________________
[gw1] linux -- Python 3.10.0 /usr/bin/python3
Traceback (most recent call last):
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/testing/plugin/pytestplugin.py", line 783, in decorate
    asyncio._run_coroutine_function(fn, *args, **kwargs)
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/testing/asyncio.py", line 33, in _run_coroutine_function
    return _util_async_run_coroutine_function(fn, *args, **kwargs)
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/util/_concurrency_py3k.py", line 155, in _util_async_run_coroutine_function
    loop = asyncio.get_event_loop()
DeprecationWarning: There is no current event loop
_____________ TextSyncDBAPI.test_sync_driver_execution[<lambda>0] ______________
[gw0] linux -- Python 3.10.0 /usr/bin/python3
Traceback (most recent call last):
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/testing/plugin/pytestplugin.py", line 783, in decorate
    asyncio._run_coroutine_function(fn, *args, **kwargs)
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/testing/asyncio.py", line 33, in _run_coroutine_function
    return _util_async_run_coroutine_function(fn, *args, **kwargs)
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/util/_concurrency_py3k.py", line 155, in _util_async_run_coroutine_function
    loop = asyncio.get_event_loop()
DeprecationWarning: There is no current event loop
_____________ TextSyncDBAPI.test_sync_driver_execution[<lambda>2] ______________
[gw0] linux -- Python 3.10.0 /usr/bin/python3
Traceback (most recent call last):
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/testing/plugin/pytestplugin.py", line 783, in decorate
    asyncio._run_coroutine_function(fn, *args, **kwargs)
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/testing/asyncio.py", line 33, in _run_coroutine_function
    return _util_async_run_coroutine_function(fn, *args, **kwargs)
  File "/builddir/build/BUILD/SQLAlchemy-1.4.15/test/../lib/sqlalchemy/util/_concurrency_py3k.py", line 155, in _util_async_run_coroutine_function
    loop = asyncio.get_event_loop()
DeprecationWarning: There is no current event loop
```

**Versions.**
 - OS: Fedora 34
 - Python: 3.10.0b1
 - SQLAlchemy: 1.4.15

**Additional context**

Downstream report: 
https://bugzilla.redhat.com/show_bug.cgi?id=1959305
**Have a nice day!**

**Comment by CaselIT** [comment]
Hi,

Thanks for reporting. That error of pytest was also what I encountered when trying to add a py3.10 gate tot the tests.

Regarding the deprecation, if I'm not reading it wrong, there is now no way of getting the existing event loop or create a new one if needed. https://docs.python.org/3.10/library/asyncio-eventloop.html?highlight=get_event_loop#asyncio.get_event_loop.

We need to call `get_running_loop` ignore the runtime error and fallback on `new_event_loop`. How thoughtful of them :\

**Comment by zzzeek** [comment]
> 
> We need to call `get_running_loop` ignore the runtime error and fallback on `new_event_loop`. How thoughtful of them :\

that should be fine no ?

**Comment by CaselIT** [comment]
sure, but seems to be overly complex for no apparent reason

**Comment by tirkarthi** [comment]
CPython bug had some discussion on how the functions will change in future : https://bugs.python.org/issue39529

**Comment by CaselIT** [comment]
@tirkarthi thanks for the context!

**Comment by CaselIT** [comment]
this was also closed by ff1d989c1433c96ac1a50c3662e77ec1025c57cb
